modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+The following DataWeave script is able to stream the XML input using each `<order-items/>` element as the streamable unit.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+output application/xml
+---
+{
+  salesorder: {
+    itemList: payload.order."order-items".*"order-item" map {
+      ("i_" ++ $$) : {
+        id: $.@id,
+        productId: $.product,
+        quantity: $.quantity,
+        price: $.price
+      }
+    }
+  }
+}
+----
+
+The script produces the following XML output.
+
+[source,XML,linenums]
+----
+<?xml version='1.0' encoding='UTF-8'?>
+<salesorder>
+  <itemList>
+    <i_0>
+      <id>31</id>
+      <quantity>2</quantity>
+      <productId>111</productId>
+      <price>8.90</price>
+    </i_0>
+  </itemList>
+  <itemList>
+    <i_1>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_1>
+  </itemList>
+  <itemList>
+    <i_2>
+      <id>31</id>
+      <quantity>2</quantity>
+      <productId>111</productId>
+      <price>8.90</price>
+    </i_2>
+  </itemList>
+  <itemList>
+    <i_3>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_3>
+  </itemList>
+  <itemList>
+    <i_4>
+      <id>31</id>
+      <quantity>7</quantity>
+      <productId>222</productId>
+      <price>5.20</price>
+    </i_4>
+  </itemList>
+</salesorder>
+----
+​
+== Validate that a Script is Streamable (Experimental Feature)
+​
+To help you code in a streamable way and avoid mistakes that can break the streaming process, DataWeave provides an _advanced, experimental_ annotation and a related directive:
+
+* `@StreamCapable()` +
+annotation: Use this annotation to validate whether the script can sequentially access a variable (typically the `payload` variable).
+* `input` directive: +
+The `@StreamCapable()` annotation requires the use of an input directive in the DataWeave script that identifies the MIME type of the data source, for example, `input payload application/xml`.
+
+The validator checks a script against the following criteria:
+
+* The variable is referenced only once
+* No index selector is set for negative access, such as `[-1]`
+* No reference to the variable is found in a nested lambda
+
+If all are true, the selected data is streamable. The following example validates successfully. The script is designed to act on the <<example_json, JSON input>> from the <<stream_json, JSON streaming>> section.
+
+[source,DataWeave,linenums]
+----
+%dw 2.0
+
+@StreamCapable()
+input payload application/json streaming=true
+output application/json
+---
+payload.family filter (member) -> member.age > 3
+----
+
+The script successfully validates and returns the following output:
+
+[source,JSON,linenums]
+----
+[
+  {
+    "name": "Pedro",
+    "age": 4
+  },
+  {
+    "name": "Matias",
+    "age": 8
+  }
+]
+----
+
+=== Validation Failures
+
+If any of the criteria that the valiator checks are false, the validation fails. 
```

Comment:
I'm not sure that this qualifies as a separate topic, with a separate title, because this intro sentence is a logical followup to "If all criteria are met" in the previous topic....
```suggestion
=== Validation Failures

If any of the criteria that the validator checks is false, the validation fails. 
```
"any (one) of is"